src/main/java/org/dice_research/opal/civet/metrics/ProviderIdentityMetric.java

```diff
@@ -0,0 +1,273 @@
+package org.dice_research.opal.civet.metrics;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.NodeIterator;
+import org.apache.jena.rdf.model.Property;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.ResIterator;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.rdf.model.SimpleSelector;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.sparql.vocabulary.FOAF;
+import org.apache.jena.vocabulary.DCAT;
+import org.apache.jena.vocabulary.DCTerms;
+import org.apache.jena.vocabulary.RDF;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dice_research.opal.civet.Metric;
+import org.dice_research.opal.common.vocabulary.Opal;
+
+/**
+ * The ProviderIdentity gives a rating to a dataset based on available
+ * informations about the dataset publisher.
+ * 
+ * @author Gourab Sahu
+ */
+public class ProviderIdentityMetric implements Metric {
+
+	public static boolean isValidURL(String checkURL) {
+		/*
+		 * Here we check whether the URL of foaf:homepage is a valid URL or not.
+		 */
+		try {
+			new URL(checkURL).toURI();
+			return true;
+		}
+
+		catch (Exception e) {
+			return false;
+		}
+	}
+
+	public void evaluatePublisher(RDFNode publisher, HashMap<String, Integer> PublisherScore) {
+
+		Resource Publisher = (Resource) publisher;
+		/*
+		 * If a publisher is of type foaf:org or foaf:person and has an non-empty name
+		 * then 5 stars are awarded.
+		 */
+		if ((Publisher.hasProperty(RDF.type, FOAF.Agent) && (!Publisher.getProperty(FOAF.name).getObject().toString().isEmpty()))) {
+
+			/*
+			 * If we do not have a publisher yet, then rate the first publisher ( of type
+			 * foaf:org /foaf:name) 5 stars.
+			 */
+			if (PublisherScore.size() == 0)
```

Comment:
If there is no publisher, avard 5 starts? Seems to be a bad implementation. Overthink.